aiidalab/app.py

```diff
@@ -228,12 +235,29 @@ def find_matching_releases(self, specifier, prereleases=None):
         matching_releases.sort(key=parse, reverse=True)
         return matching_releases
 
+    @staticmethod
+    def _strict_dependencies_met(requirements, python_bin):
+        """Check whether the given requirements are compatible with the core dependencies of a package."""
+        packages = find_installed_packages(python_bin)
+        requirements = [Requirement(r) for r in requirements]
+        requirements_dict = {r.name: r for r in requirements}
+        for core in _CORE_PACKAGES:
+            if (
+                core in requirements_dict
+                and core in packages
+                and not packages[core].fulfills(requirements_dict[core])
+            ):
+                return False
+        return True
+
     @staticmethod
     def _find_incompatibilities_python(requirements, python_bin):
         packages = find_installed_packages(python_bin)
         for requirement in map(Requirement, requirements):
-            f = [p for p in packages if p.fulfills(requirement)]
-            if not any(f):
+            pkg = packages.get(requirement.name)
```

Comment:
```suggestion
            pkg = packages.get(requirement.name, None)
```

packages is a dict so the `get` need has a default return.

Reply:
`None` is the default return value for `dict.get()` so no need to specify it I think

Reply:
> `None` is the default return value for `dict.get()` so no need to specify it I think

that's correct.

Reply:
I didn't know that 😿, thanks both!